### PR TITLE
add smpt configuration read from Environment

### PR DESCRIPTION
**Pull request by borovHokage**
http://outsourcing.nat.tepkom.ru:20080/redmine/issues/254

Task #407: Add logic to read smtp configuration from Environment.

### CODE REVIEW

File: src/MessageService/Startup.cs [hunk 1]
```diff
@@ -80,6 +80,30 @@ public void ConfigureServices(IServiceCollection services)
...
+            if (string.IsNullOrEmpty(smptEmail))
+            {
+                smptEmail = Configuration.GetConnectionString("Email");
+            }
+
+            string smptPassword = Environment.GetEnvironmentVariable("Password");
+            if (string.IsNullOrEmpty(smptPassword))
+            {
+                smptPassword = Configuration.GetConnectionString("Password");
+            }
```
Comment (KostinP) [inline src/MessageService/Startup.cs:105]: You just got the variables, what's next?

**Review by KostinP — CHANGES_REQUESTED** [review]

File: src/MessageService/Startup.cs [hunk 2]
```diff
@@ -80,6 +80,30 @@ public void ConfigureServices(IServiceCollection services)
             services.Configure<BaseRabbitMqConfig>(Configuration.GetSection(BaseRabbitMqConfig.SectionName));
             services.Configure<BaseServiceInfoConfig>(Configuration.GetSection(BaseServiceInfoConfig.SectionName));
 
+            string smptHost = Environment.GetEnvironmentVariable("Host");
+            if (string.IsNullOrEmpty(smptHost))
+            {
+                smptHost = Configuration.GetConnectionString("Host");
```
Comment (KostinP) [inline src/MessageService/Startup.cs:86]: It works?

**Review by KostinP — CHANGES_REQUESTED** [review]

File: src/MessageService/Startup.cs [hunk 3]
```diff
@@ -80,12 +80,41 @@ public void ConfigureServices(IServiceCollection services)
             services.Configure<BaseRabbitMqConfig>(Configuration.GetSection(BaseRabbitMqConfig.SectionName));
             services.Configure<BaseServiceInfoConfig>(Configuration.GetSection(BaseServiceInfoConfig.SectionName));
 
+
+            var smtpCredentialsOptions = Configuration.GetSection(SmtpCredentialsOptions.SmtpCredentials);
+
+            var smtpHost = Environment.GetEnvironmentVariable(nameof(SmtpCredentialsOptions.Host));
+            if (string.IsNullOrEmpty(smtpHost))
+            {
+                smtpCredentialsOptions[nameof(SmtpCredentialsOptions.Host)]  = smtpHost;
```
Comment (KostinP) [inline src/MessageService/Startup.cs:89]: Extra space 
...{space}{space}{=}...

File: src/MessageService/Startup.cs [hunk 4]
```diff
@@ -80,12 +80,41 @@ public void ConfigureServices(IServiceCollection services)
...
+            }
+
             string connStr = Environment.GetEnvironmentVariable("ConnectionString");
             if (string.IsNullOrEmpty(connStr))
             {
                 connStr = Configuration.GetConnectionString("SQLConnectionString");
             }
 
+            services.Configure<SmtpCredentialsOptions>(smtpCredentialsOptions);
+
```
Comment (KostinP) [inline src/MessageService/Startup.cs:117]: Double setup Configure<SmtpCredentialsOptions> (line 78, line 116)

**Review by KostinP — CHANGES_REQUESTED** [review]

File: src/MessageService/Startup.cs (hunk 4 above)
Comment (borovHokage) [reply src/MessageService/Startup.cs]: fixed

File: src/MessageService/Startup.cs (hunk 3 above)
Comment (borovHokage) [reply src/MessageService/Startup.cs]: fixed

File: src/MessageService/Startup.cs (hunk 1 above)
Comment (borovHokage) [reply src/MessageService/Startup.cs]: fixed

File: src/MessageService/Startup.cs (hunk 2 above)
Comment (borovHokage) [reply src/MessageService/Startup.cs]: fixed

File: src/MessageService/Startup.cs [hunk 5]
```diff
@@ -113,7 +107,11 @@ public void ConfigureServices(IServiceCollection services)
                 connStr = Configuration.GetConnectionString("SQLConnectionString");
             }
 
+            services.Configure<SmtpCredentialsOptions>(Configuration.GetSection(SmtpCredentialsOptions.SmtpCredentials));
             services.Configure<SmtpCredentialsOptions>(smtpCredentialsOptions);
```
Comment (KostinP) [inline src/MessageService/Startup.cs]: What do you think the `Configure` method does?

**Review by KostinP — CHANGES_REQUESTED** [review]

File: src/MessageService/Startup.cs [hunk 6]
```diff
@@ -75,17 +75,44 @@ public void ConfigureServices(IServiceCollection services)
...
             });
 
-            services.Configure<SmtpCredentialsOptions>(Configuration.GetSection(SmtpCredentialsOptions.SmtpCredentials));
-            services.Configure<TokenConfiguration>(Configuration.GetSection("CheckTokenMiddleware"));
-            services.Configure<BaseRabbitMqConfig>(Configuration.GetSection(BaseRabbitMqConfig.SectionName));
-            services.Configure<BaseServiceInfoConfig>(Configuration.GetSection(BaseServiceInfoConfig.SectionName));
+            var smtpCredentialsOptions = Configuration.GetSection(SmtpCredentialsOptions.SmtpCredentials);
+
+            var smtpHost = Environment.GetEnvironmentVariable(nameof(SmtpCredentialsOptions.Host));
+            if (string.IsNullOrEmpty(smtpHost))
```
Comment (KostinP) [inline src/MessageService/Startup.cs:81]: !

**Review by KostinP — CHANGES_REQUESTED** [review]

**Review by KostinP — APPROVED** [review]

File: src/MessageService/Startup.cs [hunk 7]
```diff
@@ -75,17 +75,44 @@ public void ConfigureServices(IServiceCollection services)
...
+            {
+                smtpCredentialsOptions[nameof(SmtpCredentialsOptions.Host)] = smtpHost;
+            }
+
+            var smtpPort = Environment.GetEnvironmentVariable(nameof(SmtpCredentialsOptions.Port));
+            if (string.IsNullOrEmpty(smtpPort))
+            {
+                smtpCredentialsOptions[nameof(SmtpCredentialsOptions.Port)] = smtpPort;
+            }
```
Comment (djino97) [inline src/MessageService/Startup.cs:90]: This can move in method other (reduce code)

**Review by djino97 — APPROVED** [review]

File: src/MessageService/Startup.cs (hunk 5 above)
Comment (borovHokage) [reply src/MessageService/Startup.cs]: fixed

File: src/MessageService/Startup.cs (hunk 6 above)
Comment (borovHokage) [reply src/MessageService/Startup.cs]: fixed

**Review by BelehovEgor — APPROVED** [review]

**Review by TIove — APPROVED** [review]

**Pull request merged**